### DISCUSSION TITLE
Easier way of inlining an SVG from a remote destination

### DISCUSSION THREAD

**Discussion by JeffersonBledsoe** [11ty/eleventy#2382] [Q&A]
My exact requirement is fetching an SVG file from my CMS and inlining that SVG through nunjucks. The current code I've got to create a shortcode for nunjucks like this:

```js
eleventyConfig.addNunjucksAsyncShortcode("inlineSvg", async url => {
    let options = {
      widths: [300],
      formats: ["svg"],
      svgShortCircuit: true
    };
    let stats = await Image(url, options);
    const svgUrl = stats.svg[0].url;

    const data = fs.readFileSync(`./${svgUrl}`, function(err, contents) {
      if (err) return err;
      return contents;
    });

    return data.toString("utf8");
  });
```

And then it's called like this:

```njk
{% inlineSvg metadata.logo.asset.url %}
```

Is there a more in-built/ simpler way of fetching an SVG from a remote an inlining it than having to pull the file down locally and re-read it?

Side-note, I'm loving 11ty!

**Comment by zachleat** [comment]
Yeah! I don’t think this made it on to the docs but if you use `dryRun: true` you’ll get a Buffer `buffer` back in the return object https://www.11ty.dev/docs/plugins/image/#dry-run

**Reply by zachleat** [reply]
See https://github.com/11ty/eleventy-img/blob/6e3c7dd1a73f3b4742a250f68d70e63c92636480/img.js#L58
